terraso_backend/apps/shared_data/visualization_tileset_tasks.py

```diff
@@ -0,0 +1,161 @@
+import csv
+import json
+import threading
+
+import pandas
+import structlog
+from django.conf import settings
+
+from apps.core.gis.mapbox import create_tileset, remove_tileset
+from apps.shared_data.services import data_entry_upload_service
+
+from .models import VisualizationConfig
+
+logger = structlog.get_logger(__name__)
+
+
+class AsyncTaskHandler:
+    def start_task(self, method, args):
+        t = threading.Thread(target=method, args=[*args], daemon=True)
+        t.start()
+
+
+def start_create_mapbox_tileset_task(visualization_id):
+    AsyncTaskHandler().start_task(create_mapbox_tileset, [visualization_id])
+
+
+def start_remove_mapbox_tileset_task(tileset_id):
+    AsyncTaskHandler().start_task(remove_mapbox_tileset, [tileset_id])
+
+
+def get_rows_from_file(data_entry):
+    type = data_entry.resource_type
+    if type.startswith("csv"):
+        file = data_entry_upload_service.get_file(data_entry.s3_object_name, "rt")
+        reader = csv.reader(file)
+        return [row for row in reader]
+    elif type.startswith("xls"):
+        file = data_entry_upload_service.get_file(data_entry.s3_object_name, "rb")
+        df = pandas.read_excel(file)
+        rows = df.values.tolist()
+        return [df.columns.tolist()] + rows
+    else:
+        raise Exception(
+            "Invalid file type for creating mapbox tileset",
+            extra={"file_type": type, "data_entry_id": data_entry.id},
+        )
+
+
+def remove_mapbox_tileset(tileset_id):
+    if tileset_id is None:
+        return
+    try:
+        remove_tileset(tileset_id)
+    except Exception as error:
+        logger.exception(
+            "Error deleting mapbox tileset",
+            extra={"tileset_id": tileset_id, "error": str(error)},
+        )
+
+
+def create_mapbox_tileset(visualization_id):
+    logger.info("Creating mapbox tileset", visualization_id=visualization_id)
```

Comment:
question: does this need to be in production? should it be logger.debug?

Reply:
@paulschreiber As right now this is implemented as an async process that the progress will not be displayed to the user, I think it is important to add these logs as info, to potentially check what happened with a tileset, from my experience I have done this not just for this type of async processes but for a lot of the functionality, so this in general feels like a good practice to me, let me know what do you think